watertap/unit_models/electrodialysis_1D.py

```diff
@@ -778,6 +1070,497 @@ def eq_recovery_mass_H2O(self, t):
+                            / self.config.property_package.charge_comp[j]
+                            for j in self.anion_set
+                        )
+                    )
+                    * log(
+                        sum(
+                            self.conc_mem_surf_mol_x[mem, "anode_right", t, x, j]
+                            for j in self.ion_set
+                        )
+                        * sum(
+                            self.concentrate.properties[t, x].conc_mol_phase_comp[
+                                "Liq", j
+                            ]
+                            for j in self.ion_set
+                        )
+                        * sum(
+                            self.conc_mem_surf_mol_x[mem, "cathode_left", t, x, j]
+                            for j in self.ion_set
+                        )
+                        ** -1
+                        * sum(
+                            self.diluate.properties[t, x].conc_mol_phase_comp["Liq", j]
+                            for j in self.ion_set
+                        )
+                        ** -1
+                    )
+                )
+            else:
+                return self.potential_nonohm_dl_x[mem, t, x] == (
+                    Constants.gas_constant
+                    * self.diluate.properties[t, x].temperature
+                    / Constants.faraday_constant
+                    * (
+                        sum(
+                            self.diluate.properties[t, x].trans_num_phase_comp["Liq", j]
+                            / self.config.property_package.charge_comp[j]
+                            for j in self.cation_set
+                        )
+                        + sum(
+                            self.diluate.properties[t, x].trans_num_phase_comp["Liq", j]
+                            / self.config.property_package.charge_comp[j]
+                            for j in self.anion_set
+                        )
+                    )
+                    * log(
+                        sum(
+                            self.conc_mem_surf_mol_x[mem, "anode_right", t, x, j]
+                            for j in self.ion_set
+                        )
+                        * sum(
+                            self.diluate.properties[t, x].conc_mol_phase_comp["Liq", j]
+                            for j in self.ion_set
+                        )
+                        * sum(
+                            self.conc_mem_surf_mol_x[mem, "cathode_left", t, x, j]
+                            for j in self.ion_set
+                        )
+                        ** -1
+                        * sum(
+                            self.concentrate.properties[t, x].conc_mol_phase_comp[
+                                "Liq", j
+                            ]
+                            for j in self.ion_set
+                        )
+                        ** -1
+                    )
+                )
```

Comment:
I am looking at the difference between what's in your if clause vs your else clause, and it's hard to tell due to the black formatting. From a glance, I am wondering why you can't condense this all rather than duplicating the expression as a whole

Reply:
Ah- concentrate and diluate concentrations and whether their sums are in the numerator or denominator-- seems to be the difference between the two log expressions. Maybe in your if-else, you could just have an alias for that ratio which changes based on the conditional, then you could just have the expression written out once with that alias in place.

Reply:
Thanks for the good suggestion.  Since this change will not actually alter the model structure or computational efficiency, plus I strongly think the overall constraint expressions may need major refinement in the next nonlinear optimization stage, can we leave this as is for now and revisit later?

Reply:
Sure! This change wouldn't really do much except reduce lines of code, so need to incorporate now.